From 8ee34ba0464df4353b027c3460383bba0e59322b Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Fri, 18 Jan 2013 11:46:26 +0900
Subject: [PATCH 36/39] wmvdec: free caps in gst_omx_wmvdec_sink_setcaps()

This patch fixes caps reference counting issue in gst_omx_wmvdec_sink_setcaps().
Without this patch, the caps reference count will be increased each time
a video is played.

To fix the issue, call gst_caps_unref() to reduce the reference count of
the caps.
---
 omx/gstomx_wmvdec.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/omx/gstomx_wmvdec.c b/omx/gstomx_wmvdec.c
index cff3398..d5eb100 100644
--- a/omx/gstomx_wmvdec.c
+++ b/omx/gstomx_wmvdec.c
@@ -104,6 +104,7 @@ gst_omx_wmvdec_sink_setcaps (GstPad * pad, GstCaps * caps)
 
   GST_INFO_OBJECT (omx_wmvdec, "setcaps (sink): %" GST_PTR_FORMAT, sink_caps);
 
+  gst_caps_unref (sink_caps);
   g_object_unref (omx_wmvdec);
 
   return result;
-- 
1.7.9.5

